service/src/main/java/bio/terra/drshub/services/DrsApiFactory.java

```diff
@@ -23,19 +16,25 @@
 @Slf4j
 public class DrsApiFactory {
 
-  private static final Integer CONNECTION_POOL_SIZE = 500;
+  private final RestTemplateFactory restTemplateFactory;
+
+  /**
+   * We create a new ApiClient for each call so that headers and tokens are not shared between
+   * calls, but each DRS provider can safely share one RestTemplate among its ApiClients.
+   */
+  private final Map<String, RestTemplate> restTemplateCache =
```

Comment:
This isn't an existing convention in the repo, but I like it and will add it here.

Reply:
I changed this back: I don't understand completely why, but this change resulted in failing tests when running the full DrsApiFactoryTest suite.

Reply:
Oh yeah, you might have to explicitly `clear()` it between tests since it's not reset like other state is. Another reason to leave it as-is I suppose.